### ISSUE TITLE
bug: `no-empty-effect` false positives only in oxlint

### ISSUE THREAD

**Issue by nickjvandyke** [nickjvandyke/eslint-plugin-react-you-might-not-need-an-effect#66]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

0.9.3

### ESLint version

9.0.0

### Describe the bug

https://github.com/nickjvandyke/eslint-plugin-react-you-might-not-need-an-effect/issues/32#issuecomment-4361554278

> FYI, I'm in the process of migrating our company's TS monorepo from eslint to oxlint and use this plugin. I'm seeing some incorrect behavior with the `no-empty-effect` rule that we aren't seeing when using through eslint.
> 
> The rule triggers only in oxlint for an effect like this:
> ```typescript
>   useEffect(() => {
>     document.getElementById('main-content').scrollIntoView();
>   }, []);
> ```
> Changing it to this satisfies the rule:
> ```typescript
>   useEffect(() => {
>     const content = document.getElementById('main-content');
>     content.scrollIntoView();
>   }, []);
> ```
> I also see these errors in oxlint for more simple cases like this:
> ```typescript
>   useEffect(() => {
>     analytics.track();
>   }, []);
> ```
> 
> Let me know if you want me to file this in a separate issue.

### Steps To Reproduce

Lint example code above.

### Expected Behavior

No warning, as in ESLint.
